test/Microsoft.AspNetCore.Server.IISIntegration.Tests/IISMiddlewareTests.cs

```diff
@@ -73,6 +75,173 @@ public class IISMiddlewareTests
         }
 
         [Fact]
+        public async Task MiddlewareShutsdownGivenANCMShutdown()
+        {
+            var assertsExecuted = false;
+            var applicationStoppingFired = false;
+            var builder = new WebHostBuilder()
+                .UseSetting("TOKEN", "TestToken")
+                .UseSetting("PORT", "12345")
+                .UseSetting("APPL_PATH", "/")
+                .UseIISIntegration()
+                .Configure(app =>
+                {
+                    var appLifetime = app.ApplicationServices.GetRequiredService<IApplicationLifetime>();
+                    appLifetime.ApplicationStopping.Register(() =>
+                    {
+                        applicationStoppingFired = true;
+                    });
+
+                    app.Run(context =>
+                    {
+                        assertsExecuted = true;
+                        return Task.FromResult(0);
+                    });
+                });
+            var server = new TestServer(builder);
+
+            var request = new HttpRequestMessage(HttpMethod.Post, "/iisintegration");
+            request.Headers.TryAddWithoutValidation("MS-ASPNETCORE-TOKEN", "TestToken");
+            request.Headers.TryAddWithoutValidation("MS-ASPNETCORE-EVENT", "shutdown");
+            var response = await server.CreateClient().SendAsync(request);
+
+            Assert.False(assertsExecuted);
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
What's the delay for? Use a ManualResetEvent with a wait timeout if you're waiting for something specific to happen.